File: webApp/src/app/common/station.ts
/**
 * 车站实体
 */
import { City } from './city';

export class Station {
  /** id */
  id: number;

  /** 车站名 */
  name: string;

  /** 城市 */
  city: City;

  constructor(data?: { id?: number, name?: string, city?: City }) {
    if (data) {
      this.id = data.id;
      this.name = data.name;
      this.city = data.city;
    }
  }
}

File: bus-ticket/src/main/java/xiaoqiangZzz/busticket/Entity/Station.java
package xiaoqiangZzz.busticket.Entity;

import com.mengyunzhi.core.entity.YunzhiEntity;

import javax.persistence.*;

@Entity
public class Station implements YunzhiEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @ManyToOne
    @JoinColumn(nullable = false)
    private City city;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public City getCity() {
        return city;
    }

    public void setCity(City city) {
        this.city = city;
    }
}
